### PR TITLE
[transform] support fuse const weight into kTransform fusion pattern

### PR DESCRIPTION
**Pull request by wyzero**
to #787

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/fusion_utils.cc [hunk 1]
```diff
@@ -552,7 +552,7 @@ FusionPatternBase::FusionPatternBase(SmallVectorImpl<Operation*>& op_list)
 // fusion pattern contains.
 int FusionPatternBase::effectiveSize() {
   return llvm::count_if(
-      op_list_, [](Operation* op) { return !matchPattern(op, m_Constant()); });
+      op_list_, [](Operation* op) { return !isa<lmhlo::ConstantOp>(op); });
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/fusion_utils.cc:555]: why is it narrowed to `lmhlo::ConstantOp`?

File: tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc [hunk 2]
```diff
@@ -45,21 +42,49 @@ bool TransformBasedCpuFusionStrategy::isFusible(Operation* op) {
...
-  fusionPattern.setFusionType(inferredFusionType);
+
+  DenseSet<Value> dotWeights;
+  DenseSet<Operation*> supportedDotOps;
+  for (Operation* op : fusionPattern.getOpList()) {
+    // early return for the case where there are non supported ops.
+    if (!this->isFusible(op)) return true;
+    if (isSupportedDot(op)) {
+      supportedDotOps.insert(op);
+      dotWeights.insert(op->getOperand(1));
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc:72]: nit: No need to check if it's a constant op? could operand 0 be constant (seems unusual ...)?

File: tao_compiler/mlir/disc/transforms/fusion_utils.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/fusion_utils.cc:555]: m_Constant only works for const on tensor level.

File: tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc (hunk 2 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc:72]: the producer of `op->getOperand(1)` is usually a alloc op, not const op. The const op is also one consumer of `op->getOperand(1)`. We this at line 79.

Comment (qiuxiafei) [reply tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc:72]: got it.

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**